assets/js/blocks/classic-shortcode/index.tsx

```diff
@@ -56,31 +56,44 @@ const conversionConfig: { [ key: string ]: BlockifiedTemplateConfig } = {
 	fallback: blockifiedFallbackConfig,
 };
 
-const pickBlockClientIds = ( blocks: Array< BlockInstance > ) =>
-	blocks.reduce< Array< string > >( ( acc, block ) => {
-		if ( block.name === 'core/template-part' ) {
-			return acc;
-		}
+const pickBlockClientId = ( blocks: Array< BlockInstance > ) => {
+	const groupBlock = blocks.find(
+		( block ) =>
+			block.name === 'core/group' &&
+			block.innerBlocks.some(
+				( innerBlock ) =>
+					innerBlock.name === 'woocommerce/cart' ||
+					innerBlock.name === 'woocommerce/checkout'
+			)
+	);
 
-		return [ ...acc, block.clientId ];
-	}, [] );
+	if ( groupBlock ) {
+		return (
+			groupBlock.innerBlocks.find( ( block ) => {
+				return (
+					block.name === 'woocommerce/cart' ||
+					block.name === 'woocommerce/checkout'
+				);
+			} )?.clientId || ''
+		);
+	}
+};
```

Comment:
Isn't this logic already covered by the groupBlock logic?

Reply:
Both loops are needed; first to find the parent group block, and 2nd to select the client ID of the cart/checkout block within it.

I've improved the code slightly in https://github.com/woocommerce/woocommerce-blocks/pull/11228/commits/b16cd4c131b537a3f6eee6df7f54da837bdc795a to support cases where a group is not present.